pkg/operator/ceph/object/s3-handlers.go

```diff
@@ -69,11 +79,29 @@ func NewS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byt
 			WithLogLevel(logLevel),
 	)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "failed to create v1 session")
+	}
+	v1Client := s3.New(v1Session)
+
+	// -----------------------------
+	// SDK v2 client initialization
+	// -----------------------------
+	v2Config := v2aws.Config{
+		Region: CephRegion,
+		Credentials: v2aws.NewCredentialsCache(
+			v2creds.NewStaticCredentialsProvider(accessKey, secretKey, ""),
+		),
+		HTTPClient:   httpClient,
+		BaseEndpoint: v2aws.String(endpoint),
 	}
```

Comment:
I don't see all the same options from the V1 config here. Importantly, TLS is missing. Log level is also missing, which will be important for debugging. Some kind of "force path style" may also be needed because a default RGW deploy doesn't support vhost style addressing. Max retries may also be important, just so we aren't waiting around too long, but not sure about that one.